go/api_default.go

```diff
@@ -144,32 +182,296 @@ func getDashboardByName(name string, grafanaUrl string, grafanaApiKey string) (u
 
 	dash := g[0]
 
-	url = grafanaUrl + dash.Url
+	url = grafanaURL + dash.Url
 	id = dash.Id
 	uid = dash.Uid
 	version = dash.Version
 
 	return url, id, uid, version, found, err
 }
 
+func generatePassword() (passwd string, err error) {
+	passwd, err = password.Generate(18, 9, 9, false, false)
+	if err != nil {
+		return
+	}
+	return passwd, nil
+}
+
+func createUser(user User, grafanaURL string, auth string) (createdUser User, err error) {
+	existingUser, found, err := getUserByEmail(user.Email, grafanaURL, auth)
+
+	if err != nil {
+		return
+	}
+
+	if found {
+		logrus.Infof("User %s exists already", user.Email)
+		return existingUser, nil
+	}
+
+	if user.Password == "" {
+		passwd, err := generatePassword()
+		if err != nil {
+			return user, err
+		}
+		user.Password = passwd
+	}
+
+	// We need email to tie user back to the hub
+	if user.Email == "" {
+		logrus.Infof("User create request received without an email")
+		err = errors.New("Email is a required field")
+		return user, err
+	}
+
+	// Can only specify email or username in Grafana create user method, not both
+	if user.Username != "" {
+		user.Username = ""
+	}
+
+	payload, _ := json.Marshal(user)
```

Comment:
:-O ... no error checker .. how could you dan!!!

Reply:
_ = @gambol99